aws/resource_aws_route53_query_log.go

```diff
@@ -0,0 +1,91 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53"
+)
+
+func resourceAwsRoute53QueryLog() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsRoute53QueryLogCreate,
+		Read:   resourceAwsRoute53QueryLogRead,
+		Delete: resourceAwsRoute53QueryLogDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cloudwatch_log_group_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+
+			"zone_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsRoute53QueryLogCreate(d *schema.ResourceData, meta interface{}) error {
+	r53 := meta.(*AWSClient).r53conn
+
+	input := &route53.CreateQueryLoggingConfigInput{
+		CloudWatchLogsLogGroupArn: aws.String(d.Get("cloudwatch_log_group_arn").(string)),
+		HostedZoneId:              aws.String(d.Get("zone_id").(string)),
+	}
+
+	log.Printf("[DEBUG] Creating Route53 query logging configuration: %#v", input)
+	out, err := r53.CreateQueryLoggingConfig(input)
+	if err != nil {
+		return fmt.Errorf("Error creating Route53 query logging configuration: %s", err)
+	}
+	log.Printf("[DEBUG] Route53 query logging configuration created: %#v", out)
```

Comment:
Nitpick, but all AWS SDK structs have reasonable `String()` implementation, so it's safe (and usually better) to just use `%s`.